captum/metrics/_core/infidelity.py

```diff
@@ -0,0 +1,314 @@
+import torch
+
+from ..._utils.common import (
+    ExpansionTypes,
+    _expand_additional_forward_args,
+    _expand_target,
+    _format_additional_forward_args,
+    _format_input,
+    _format_tensor_into_tuples,
+    _is_tuple,
+    _run_forward,
+)
+
+
+def infidelity_attr(
+    forward_func,
+    perturb_func,
+    inputs,
+    attributions,
+    additional_forward_args=None,
+    target=None,
+    n_samples=10,
+    max_examples_per_batch=None,
+):
+    r"""
+    Explanation infidelity represents the expected mean-squared error
+    between the explanation multiplied by a meaningful input perturbation
+    and the differences between the predictor function at its input
+    and perturbed input.
+    More details about the measure can be found in the following paper:
+    https://arxiv.org/pdf/1901.09392.pdf
+
+    It is derived from the completeness property of well-known attribution
+    algorithms and is an optimal and generalized notion of Sensitivy-n.
+    The latter measures correlations between the sum of the attributions
+    and the differences of the predictor function at its input and
+    fixed baseline. More details about the Sensitivity-n can be found here:
+    https://arxiv.org/pdf/1711.06104.pdfs
+
+    The users can perturb the inputs any desired way by providing any
+    perturbation function that takes the inputs and returns perturbed
+    inputs and corresponding perturbations.
+
+    This specific implementation is primarily tested for attribution-based
+    explanation methods but the idea can be expanded to use for non
+    attribution-based interpretability methods as well.
+
+    Args:
+
+        forward_func (callable):
+                The forward function of our model or any modification of it.
+        perturb_func (callable):
```

Comment:
I actually, thought about it but by allowing to return perturbations and perturbed inputs independently we give users have more flexibility. They can do :
```

perturbed_input = input [operation] perturbation
```
In this case [operation] can be add, subtract, multiply or anything else.

Reply:
I think a better example of this is when the perturbation `I` is binary. As explained in the paper. For the global explanations the authors used binary perturbation. In that case `I` is like a mask and in (X - I) is basically the input with ablated features where I is 0.

Reply:
I can understand that the users might use addition for example and that can be confusing. In general perturbation can be negative but in that can the negative values would be reflected also in the multiplier `I`.
By taking perturbed input do you mean returning only perturbed input in `perturb_func `?

The problem with only taking perturbed input in case of global is that if the ablation value is not 0 how will we compute the ablation mask ? In that case the user needs to provide `I*Attribution` which would be a bit confusing?

We can potentially document current behavior and also perform asserts in case of local explanation.

Reply:
Actually, I guess we can generate the mask based on the differences probably.